assets/course-builder/frontend.js

```diff
@@ -0,0 +1,40 @@
+import { render, useEffect, useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+import Lesson from './lesson';
+import '../shared/data/api-fetch-preloaded-once';
+
+const element = document.getElementById( 'course-outline-block' );
+
+const CourseOutlineFrontend = () => {
+	const [ lessons, setLessons ] = useState( [] );
+	const [ loading, setLoading ] = useState( false );
+	useEffect( () => {
+		setLoading( true );
+		apiFetch( {
+			path: `/sensei-internal/v1/course-builder/course-lessons/${ element.dataset.id }`,
+		} ).then( ( l ) => {
+			setLessons( l );
+			setLoading( false );
+		} );
+	}, [] );
+
+	if ( loading ) {
+		return 'Loading API...';
```

Comment:
This loading appears while loading the API. But as we're using the cache, the Promise will be solved immediately.